client/src/pages/automation/project/utils/getNestedObject.ts

```diff
@@ -0,0 +1,25 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const getNestedObject = (jsonObj: any, selector: string) => {
+    const selectors = selector.split('.');
+
+    return selectors.reduce((obj, key) => {
+        let finalKey: string | undefined = key;
+        let index = -1;
+
+        if (finalKey === '[index]') {
+            index = 0;
+            finalKey = undefined;
+        } else if (finalKey.endsWith('[index]')) {
+            index = 0;
+            finalKey = finalKey.substring(0, finalKey.length - '[index]'.length);
```

Comment:
ja bi i ovaj `[index]` isto posla kao argument, da kod bude robustniji.

Reply:
@kresimir-coko to ću tebi prepustiti kasnije, zasada je ovo jedini case